packages/loader/container-loader/src/container.ts

```diff
@@ -1533,6 +1540,42 @@ export class Container
 
 		let opsBeforeReturnP: Promise<void> | undefined;
 
+		if (loadMode.pauseAfterLoad === true) {
+			// If we are trying to pause at a specific sequence number, ensure the latest snapshot is not newer than the desired sequence number.
+			if (loadMode.opsBeforeReturn === "sequenceNumber") {
+				assert(loadToSequenceNumber !== undefined, "sequenceNumber should be defined");
+				if (this.deltaManager.lastSequenceNumber > loadToSequenceNumber) {
+					throw new Error(
+						"Cannot satisfy request to pause the container at the specified sequence number. Most recent snapshot is newer than the specified sequence number.",
+					);
+				}
+			}
+
+			// Force readonly mode - this will ensure we don't receive an error for the lack of join op
+			this.forceReadonly(true);
+
+			// We need to setup a listener to stop op processing once we reach the desired sequence number (if specified).
+			const opHandler = (message: ISequencedDocumentMessage) => {
+				if (loadToSequenceNumber === undefined) {
+					// If there is no specified sequence number, pause after the inbound queue is empty.
+					if (this.deltaManager.inbound.length !== 0) {
+						return;
+					}
```

Comment:
If the user wants to pause, but does not specify a sequence number, then we need to pick a point to pause. Here I chose after the initial inbound queue is empty. Does this seem appropriate?

Reply:
Seems reasonable to me!